strider/fetcher.py

```diff
@@ -61,14 +62,20 @@ async def lookup(
         self,
         qgraph: Graph = None,
         call_stack: List = [],
+        qid: str = "",
     ):
         """Expand from query graph node."""
+        if qid:
+            qid = qid + "." + str(uuid.uuid4())[:8]
+        else:
+            qid = str(uuid.uuid4())[:8]
         # if this is a leaf node, we're done
         if qgraph is None:
             qgraph = Graph(self.qgraph)
         if not qgraph["edges"]:
-            self.logger.info(f"Finished call stack: {(', ').join(call_stack)}")
-            yield QueryGraph.parse_obj(
+            self.logger.info(f"[{qid}] Finished call stack: {(', ').join(call_stack)}")
+            # gets sent to generate_from_result for final result merge and then yield to server.py
+            yield KnowledgeGraph.parse_obj(
```

Comment:
Yeah, QueryGraph was always wrong, though it didn't matter because they both have `nodes` and `edges`. If you look down in `generate_from_result`, self.lookup is supposed to yield a subkgraph. I just updated this to return the technically correct type.